packages/cubejs-schema-compiler/adapter/BaseTimeDimension.js

```diff
@@ -48,8 +48,10 @@ class BaseTimeDimension extends BaseFilter {
     return super.aliasName();
   }
 
-  unescapedAliasName() {
-    return `${this.query.aliasName(this.dimension)}_${this.granularity || 'date'}`; // TODO date here for rollups
+  unescapedAliasName(context = {}) {
```

Comment:
Context is a thread local so it's safe to access it right from query everywhere during query generation. It's thread local solely for the purpose of not polluting arguments of methods. So we shouldn't add context to arguments everywhere in the code. Otherwise every method should have context argument.

Reply:
it's truly a bad practice to use `context` from the arguments.

My purpose is to distinguish `dimensionSQL()` from `selectColumns()` and  return the corresponding SQL.
If we access the `context` by `safeEvaluateSymbolContext()` rather than `arguments`, the SQL will become `date_trunc('month', ("orders.created_at_date)") "orders.created_at_date"`.

Maybe you have some elegant solutions for it?

Reply:
@Justin-ZS Ah. Gotcha. That's right. We'd need to go another route then. I believe we can apply following change:
```
    if (context.rollupQuery) {
      if (context.rollupGranularity === this.granularity) {
        return super.dimensionSql();
      } else {
        return this.query.timeGroupedColumn(
          this.query.safeEvaluateSymbolContext().granularityOverride || this.granularity,
          this.query.escapeColumnName(this.unescapedAliasName(context.rollupGranularity))
        );
      }
    }
```
And add explicit `rollupGranularity` argument to `unescapedAliasName()` rather than `context`.

Reply:
@Justin-ZS Do you need any additional help on this one? Seems this one is only thing we need to fix and we're good to go.

Reply:
@paveltiunov 
Thank you! I will ask for some help if I stuck somewhere.
I just have not to start to fix this yet.
This weekend, I will take some time to update it and complete the pull request.